data/transactions/logic/eval.go

```diff
@@ -612,26 +619,243 @@ func (cx *EvalContext) RunMode() RunMode {
 	return cx.runModeFlags
 }
 
-// StackType describes the type of a value on the operand stack
-type StackType byte
+// avmType describes the type of a value on the operand stack
+type avmType byte
 
 const (
-	// StackNone in an OpSpec shows that the op pops or yields nothing
-	StackNone StackType = iota
+	// avmNone in an OpSpec shows that the op pops or yields nothing
+	avmNone avmType = iota
 
-	// StackAny in an OpSpec shows that the op pops or yield any type
-	StackAny
+	// avmAny in an OpSpec shows that the op pops or yield any type
+	avmAny
 
-	// StackUint64 in an OpSpec shows that the op pops or yields a uint64
-	StackUint64
+	// avmUint64 in an OpSpec shows that the op pops or yields a uint64
+	avmUint64
 
-	// StackBytes in an OpSpec shows that the op pops or yields a []byte
-	StackBytes
+	// avmBytes in an OpSpec shows that the op pops or yields a []byte
+	avmBytes
 )
 
+func (at avmType) String() string {
+	switch at {
+	case avmNone:
+		return "none"
+	case avmAny:
+		return "any"
+	case avmUint64:
+		return "uint64"
+	case avmBytes:
+		return "[]byte"
+	}
+	return "internal error, unknown type"
+}
+
+var (
+	// StackUint64 is any valid uint64
+	StackUint64 = NewStackType(avmUint64, bound(0, math.MaxUint64))
+	// StackBytes is any valid bytestring
+	StackBytes = NewStackType(avmBytes, bound(0, maxStringSize))
+	// StackAny could be Bytes or Uint64
+	StackAny = StackType{
+		Name:        avmAny.String(),
+		AVMType:     avmAny,
+		ValueBound:  StackUint64.ValueBound,
+		LengthBound: StackBytes.LengthBound,
+	}
+	// StackNone is used when there is no input or output to
+	// an opcode
+	StackNone = StackType{
+		Name:    avmNone.String(),
+		AVMType: avmNone,
+	}
+
+	// StackBoolean constrains the int to 1 or 0, representing True or False
+	StackBoolean = NewStackType(avmUint64, bound(0, 1), "bool")
+	// StackAddress represents an address
+	StackAddress = NewStackType(avmBytes, static(32), "addr")
+	// StackBytes32 represents a bytestring that should have exactly 32 bytes
+	StackBytes32 = NewStackType(avmBytes, static(32), "[32]byte")
+	// StackBigInt represents a bytestring that should be treated like an int
+	StackBigInt = NewStackType(avmBytes, bound(0, maxByteMathSize), "bigint")
+	// StackMethodSelector represents a bytestring that should be treated like a method selector
+	StackMethodSelector = NewStackType(avmBytes, static(4), "method")
+	// StackStorageKey represents a bytestring that can be used as a key to some storage (global/local/box)
+	StackStorageKey = NewStackType(avmBytes, bound(0, 64), "key")
+	// StackBoxKey represents a bytestring that can be used as a key to a box
+	StackBoxKey = NewStackType(avmBytes, bound(1, 64), "bkey")
+
+	// AllStackTypes is a list of all the stack types we recognize
+	// so that we can iterate over them in doc prep
+	AllStackTypes = []StackType{
```

Comment:
I don't see this being used, but agree we need a new table in the docs describing these types.

Reply:
I think this is used in a follow up PR, left from when I split it into multiple PRs. Meant to make the full list of stack types available to the `opdoc` tool

Reply:
If getting this doc table into this PR isn't too hard, I'd much prefer it. I don't like the spec being in a state in which it uses undefined terms.

I also think there are several types here you won't document though, like  `StackBytes32` and `StackZeroBytes`. If that's true, is there any other reason for them to exist as Go consts? Maybe they they are used when parsing `proto`. At any rate, I don't know that these consts need to be Go exported. `opdoc` will just take `AllStackTypes` and use what's in there, never actually using the Go names.

Reply:
I've added a new markdown file `named_stack_types.md` to the opdoc generation and added it to `README_in.md`